PLIER_pathways_analysis.Rmd

```diff
@@ -0,0 +1,76 @@
+---
+title: "PLIER pathways analysis"
+output: html_notebook
+---
+
+```{r}
+meaningful_difference <- 0.2
+```
+
+### What additional pathways are identified by PLIER after doubling the sample size?
+
+Here we look for cancer pathways that are found more frequently in the data with the full sample size than in data with half the sample size.
+We ran PLIER 10 times using each combination of data normalization method and sample size (half or full).
+Out of those 10 runs, we found the proportion of times each pathway was significantly associated with a latent variable.
+We set an arbitrary "meaningful difference" threshold of `r meaningful_difference` to detect when that proportion was meaningfully greater using full sample size compared to half sample size and require array and RNA-seq data to both to satisfy the condition.
+We also arbitrarily limit results to those pathways found in over half of runs using the full size data.
+
+### Load packages and pathways
+
+```{r}
+library(tidyverse)
+library(PLIER)
+
+data(bloodCellMarkersIRISDMAP)
+data(canonicalPathways)
+data(oncogenicPathways)
+data(svmMarkers)
+```
+
+### Read in data for each cancer type
+```{r}
+brca_pathways_df <- read_tsv("plots/data/BRCA_subtype_PLIER_pathways.tsv") %>%
+  mutate(cancer_type = "BRCA")
+gbm_pathways_df <- read_tsv("plots/data/GBM_subtype_PLIER_pathways.tsv") %>%
+  mutate(cancer_type = "GBM")
+```
+
+### Filter data to identify pathways detected more frequently in full data
+```{r}
+
+bind_rows(brca_pathways_df,
+          gbm_pathways_df) %>%
+  filter(FDR < 0.05) %>%
+  group_by(nmeth, pseq, pathway, cancer_type) %>%
+  summarize(n_seeds = length(unique(seed_index)),
+            prop_seeds = n_seeds/10,
+            .groups = "drop") %>%
+  mutate(nmeth = str_remove_all(nmeth, "[\\(\\)]")) %>%
+  mutate(nmeth = str_replace_all(nmeth, c(" " = "_", "-" = "_"))) %>%
+  mutate(nmeth_pseq = str_c(nmeth, pseq, sep = "_")) %>%
+  select(cancer_type, pathway, prop_seeds, nmeth_pseq) %>%
+  pivot_wider(id_cols = c("cancer_type", "pathway"),
+              names_from = nmeth_pseq,
+              values_from = prop_seeds) %>%
+  mutate_if(is.numeric, replace_na, 0) %>%
+  filter(log_0 - array_only_50 >= meaningful_difference,
+         log_100 - seq_only_50 >= meaningful_difference,
+         log_0 > 0.5,
+         log_100 > 0.5,
+         npn_50 > 0.5) %>%
+  select(cancer_type, pathway, array_only_50, seq_only_50, log_0, log_100, npn_50) %>%
+  filter(pathway %in% colnames(oncogenicPathways)) %>%
+  arrange(cancer_type) %>%
```

Comment:
Would love to see more comments throughout this long chain

Reply:
Commented!